framework/include/dirackernels/DiracKernel.h

```diff
@@ -216,6 +216,9 @@ class DiracKernel :
   /// Derivative of u_dot wrt u
   const VariableValue & _du_dot_du;
 
+  /// fully consider duplicate points
+  const bool _allow_duplicate_points;
```

Comment:
I'd rather go for the opposite, `drop_duplicate_points`. Again, I don't really want to expose "multiplicity" to the user. For the user _not dropping duplicate points_ is the same thing as "multiplying contribution by point multiplicity". Only the latter sound much more complicated IMO.